File: 0199. Binary Tree Right Side View/Binary Tree Right Side View.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    private void dfs(TreeNode node, List<Integer> res, int depth) {
        if (node == null) return;
        
        if (res.size() == depth) {
            res.add(node.val);
        }
        
        dfs(node.right, res, depth + 1);
        dfs(node.left, res, depth + 1);
    }
    
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        dfs(root, res, 0);
        return res;
    }
}

File: 0199. Binary Tree Right Side View/Binary Tree Right Side View.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def dfs(self, node: TreeNode, res: List[int], depth: int) -> None:
        if node is None:
            return
        if len(res) == depth:
            res.append(node.val)
            
        self.dfs(node.right, res, depth + 1)
        self.dfs(node.left, res, depth + 1)
        
    def rightSideView(self, root: TreeNode) -> List[int]:
        res = list()
        self.dfs(root, res, 0)
        return res